doc/flame/rendering/text_rendering.md

```diff
@@ -113,6 +113,63 @@ You can find all the options under [TextBoxComponent's
 API](https://pub.dev/documentation/flame/latest/components/TextBoxComponent-class.html).
 
 
+### ElementComponent
+
+If you want to render an arbitrary Element, ranging from a single TextElement to a formatted
```

Comment:
Is it arbitrary, or is it only related to text? Because if it is only regarding text I think it could have a more explanatory name?

Reply:
By arbitrary I mean any child of Element, not only TextElement. That is how the classes are already named, even though they all have to do with the rendering of text. I didn't like calling it ElementComponent either but TextElementComponent would definitely be misnomer.

Reply:
Something completely different then? `TextRenderingComponent` or `RichTextComponent` or something? `ElementComponent` doesn't tell the user much about what it does. 😅

Reply:
Honestly I feel `Element` is already a bad name for those reasons exactly; could be named TextElement and what is currently TextElement could be TextLineElement. But might be too much of a breaking change to change that now, probably?

For the component, I am over the fence because while I see just calling it ElementComponent can cause confusion it is actually clearly describing what it is (the confusing being on the other class, Element). `TextRenderingComponent` or `RichTextComponent` can also be confusing because are hiding what it actually does. Like unclear what the difference between `TextRenderingComponent` and `TextComponent` would be just from the name. So maybe `RichTextComponent` then?

Reply:
actually never mind all that. I think it is worth it and not that bad to rename all these classes. let me put out a proposal.

Reply:
https://github.com/flame-engine/flame/pull/2700

Reply:
then this will become `TextElementComponent` and bliss will befall upon us both 😌

Reply:
done!